src/hotspot/share/opto/node.cpp

```diff
@@ -1787,10 +1787,7 @@ void Node::dump(const char* suffix, bool mark, outputStream *st) const {
   if (t != NULL && (t->isa_instptr() || t->isa_klassptr())) {
     const TypeInstPtr  *toop = t->isa_instptr();
     const TypeKlassPtr *tkls = t->isa_klassptr();
-    ciKlass*           klass = toop ? toop->klass() : (tkls ? tkls->klass() : NULL );
-    if (klass && klass->is_loaded() && klass->is_interface()) {
-      st->print("  Interface:");
-    } else if (toop) {
+    if (toop) {
```

Comment:
Why did you remove the interface case?

Reply:
I brought it back as this doesn't really matter for this PR. The next change in the series removes the klass() accessor and that creates all sort of issues. In some cases where the use of klass() seems to have little value (like here, after all interfaces can't be trusted), I removed the code using the klass() rather than try to fix it.